kotlinx-coroutines-core/common/src/flow/operators/Lint.kt

```diff
@@ -186,3 +186,15 @@ public suspend inline fun <T> SharedFlow<T>.toSet(destination: MutableSet<T>): N
 @InlineOnly
 public suspend inline fun <T> SharedFlow<T>.count(): Int =
     (this as Flow<T>).count()
+
+/**
+ * @suppress
+ */
+@Suppress("DeprecatedCallableAddReplaceWith")
+@Deprecated(
+    level = DeprecationLevel.WARNING,
+    message = "SharedFlow never completes, so this terminal operation never completes. Use 'replayCache.last()' for reading the last emitted item in the replay cache.",
```

Comment:
One use case for `sharedFlow.last()` that comes to mind is to subscribe to a shared flow so that it "starts working" (as shared flows can be configured behave differently depending on whether there are subscribers: https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/-sharing-started/), but doesn't actually do anything. In other words, subscribing for the side effects of subscribing and not for the value. In that scenario, the idiomatic approach is to call `sharedFlow.collect()`. This could be worth highlighting in the replacement message.

Reply:
This line is too long: we try to stick to 120 characters per line, and typically, we don't even fill that limit to the brim, instead breaking strings on punctuation (dots, commas...). Example: https://github.com/Kotlin/kotlinx.coroutines/pull/4433/files#diff-c6a5519f3b99094bab5ee5f1797928d8ed5df48460d0840e642bcf7da495a6dbR61-R64 The reason is to avoid large git diffs when adding some new words in the middle of the text and causing a cascade of changes due to reformatting.